src/trackers/helpers/readers.js

```diff
@@ -0,0 +1,97 @@
+const fs = require('fs')
+const {gunzip} = require('zlib')
+const {Client} = require('pg')
+const Cursor = require('pg-cursor')
+const {promisify} = require('util')
+
+const dogunzip = promisify(gunzip)
+
+class JSONFileDataReader {
+    constructor(path) {
+        this.path = path
+        this._fileList = null
+    }
+
+    async getFileList() {
+        if (!this._fileList) {
+            this._fileList = await fs.promises.readdir(this.path)
+        }
+        return this._fileList
+    }
+
+    async length() {
+        return (await this.getFileList()).length - 1
+    }
+
+    async *iterator() {
+        for (const file of await this.getFileList()) {
+            // transparently expand gzipped files
+            const fileData = file.endsWith('.gz')
+                ? (await dogunzip(await fs.promises.readFile(`${this.path}/${file}`))).toString('utf-8')
+                : fs.promises.readFile(`${this.path}/${file}`, 'utf8')
+            const siteData = JSON.parse(fileData)
+            if (siteData.initialUrl) {
```

Comment:
Does it happen that this is false? Should we at least print a message that the file was broken?

Reply:
Previously this check was in `processSite`, and serves the purpose of filtering files in the data folder which don't contain data for a site, namely `metadata.json`.

Reply:
ah gotcha, we should probably change that at some point to do the check based on the filename and not existence of this field, but lets leave it out of this PR 👍